packages/flutter/lib/src/material/dialog.dart

```diff
@@ -950,6 +950,85 @@ Widget _buildMaterialDialogTransitions(BuildContext context, Animation<double> a
 /// Returns a [Future] that resolves to the value (if any) that was passed to
 /// [Navigator.pop] when the dialog was closed.
 ///
+/// ### State Restoration in Dialogs
+///
+/// Using this method with not enable state restoration for the dialog. In order
```

Comment:
with -> will

(I wish we could share the docs with the other locations, but looks like they are slightly different)